Src/VimCore/Modes_Insert_Components.fs

```diff
@@ -157,6 +152,7 @@ type internal TextChangeTracker
     member x.EffectiveChange =
         match _effectiveChangeData with
         | Some data ->
+            Contract.Assert(data.Snapshot = _textView.TextSnapshot)

```

Comment:
Believe this will be false in some extreme corner cases. The implementation of `ITextView.TextSnapshot` doesn't directly call into `ITextBuffer.CurrentSnapshot` but rather uses a cached field. This cached field is only updated after `PerformLayout` occurs. In cases where `PerformLayout` is delayed, or this is called on an `ITextBuffer` that maps to multiple `ITextView` I'm pretty sure you could observe this failing. 

This should be safer: `_textView.TextBuffer.CurrentSnapshot`

Reply:
Note: I'm not sure if what I wrote is true for every version of VS. For 2017 at least though this is true (just looked at the implementation in ILSpy)

---
In reply to: [235755886](https://github.com/jaredpar/VsVim/pull/2426#discussion_r235755886) [](ancestors = 235755886)